org.gnome.Evolution.Extension.evolution-etesync.json

```
@@ -0,0 +1,91 @@
+{
+	"id": "org.gnome.Evolution.Extension.evolution-etesync",
+	"runtime": "org.gnome.Evolution",
+	"branch": "stable",
+	"runtime-version": "stable",
+	"sdk": "org.gnome.Sdk//44",
+        "sdk-extensions" : [
+            "org.freedesktop.Sdk.Extension.rust-stable"
+        ],
+	"build-extension": true,
+	"separate-locales": false,
+	"appstream-compose": false,
+	"finish-args": [],
+	"build-options" : {
+		"prefix": "/app/evolution/extensions/evolution-etesync",
+		"prepend-path": "/app/evolution/extensions/evolution-etesync",
+		"prepend-pkg-config-path": "/app/evolution/extensions/evolution-etesync/lib/pkgconfig",
+		"append-path": "/usr/lib/sdk/rust-stable/bin",
+		"cflags": "-O2 -g -Wno-deprecated-declarations",
+		"cxxflags": "-O2 -g -Wno-deprecated-declarations"
+	},
+        "cleanup": [
+            "/include",
+            "/lib/cmake",
+            "/lib/pkgconfig",
+            "/share/pkgconfig",
+            "/share/aclocal",
+            "*.la",
+            "*.a"
+        ],
+	"modules": [
+		{
+			"name": "libetebase",
+			"buildsystem": "simple",
+			"build-commands": [
+				"mkdir .cargo",
+				"cp cargo/config .cargo/",
+				"mv libetebase-Cargo.lock Cargo.lock",
+				"cargo --offline fetch --manifest-path Cargo.toml",
+				"cargo --offline build --release",
+				"PREFIX=/app/evolution/extensions/evolution-etesync make V=1 VERBOSE=1 pkgconfig",
+				"PREFIX=/app/evolution/extensions/evolution-etesync make V=1 VERBOSE=1 install",
+				"mv /app/evolution/extensions/evolution-etesync/lib/libetebase.so /app/evolution/extensions/evolution-etesync/lib/libetebase.so.0",
+				"ln -s /app/evolution/extensions/evolution-etesync/lib/libetebase.so.0 /app/evolution/extensions/evolution-etesync/lib/libetebase.so"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://github.com/etesync/libetebase/archive/v0.5.3.tar.gz",
+					"sha256": "78c79f72df40ae4ed85beca593d870159f923e8b9e0111c16e34b6142c634995"
+				},
+				{
+					"type": "file",
+					"path": "libetebase-Cargo.lock"
+				},
+				"libetebase-cargo-sources.json"
+			]
+		},
+
+		"shared-modules/intltool/intltool-0.51.json",
+
+		{
+			"name": "evolution-etesync",
+			"buildsystem": "cmake-ninja",
+			"config-opts": [
+				"-DFORCE_INSTALL_PREFIX=/app/evolution/extensions/evolution-etesync"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://download.gnome.org/sources/evolution-etesync/1.1/evolution-etesync-1.1.1.tar.xz",
+					"sha256": "ae89773450caf25328825576bc9a675ce885e1c775481130c98f89e18555d3a3"
+				}
+			]
+		},
+
+		{
+			"name": "metainfo",
+			"buildsystem": "simple",
+			"build-commands": [
+				"install -t $FLATPAK_DEST/share/app-info/xmls/ -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml"
```

```suggestion
				"install -t $FLATPAK_DEST/share/metainfo -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml"
```


you never ever install anything in `app-info`.

---

you also need to run the appstream-compose by hand like any other extension.

---

`"appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}"`